### DISCUSSION TITLE
Drawing contours at significant changes in elevation

### DISCUSSION THREAD

**Discussion by stefankinell** [karttapullautin/karttapullautin#159] [General]
One thing I notice when I am doing a competition map for orienteering and compare the curves from karttapullautin with the ones I draw as a map maker in the field is that KP has a tendencies to draw formlines where I would in real life draw a contour, and that the contours then don't end up where the real significant changes in elevation take place.

As a map maker you try to draw the contour where you have significant changes in the nature. You let the contour glide up or down vs where the real five meters line is to let it show what you actually see in nature. The norm says we are allowed to glide 25% (or is it 20%) but most professional map makers says that is more of a guideline, the important thing is to depict what you see in nature. 

If I remember an old conversation with Jarkko correct, KP does gliding of the contours to some extent, it does not depict the exact meter contorus. But I am not able to comprehend the logics of the code to understand. Those of you who go have been diving more into the code - can you describe what KP actually does? WHen looking at my current mapping project there seems to be very small differences between contours and my underlaying 0,3 m contours - which implies that the concours does not glide, only that they are smoother than the 0,3 m contours.

If there is no gliding implemented - would that be possible to achive? If it is alreday in place -is it possible to let KP glide even more, so that the contours end up where it is really significant changes in nature? Perhaps as default, perhaps as a variable - to set how much you allow KP to deviate with a contour from the actual five meters steps (or what you have specified). Would this be feasible to implement?

Where we are now with good level of speed improvements - this would be a reall next game changer for a tool like karttapullautin.

**Comment by Jagge** [comment]
Controus are "glided" near knolls. But not for slopes without any knolls. The purpose is not failing to map signigicant knolls for them happening to fall horisontally between contour interval. And for slopes contour interval + form line is just thought to be enough to capture the landform well enough, the issue of lanform falling between vertically between controus is not as bad as flat areas with knolls.  Anyeay, to glide contours there you would need to detect the need somehow and implement the change without causing any too bad side effect - quite easily you tweaking like this makes it look bad around the edges of the tweaked part.

**Reply by stefankinell** [reply]
I understand. I see it from two perspectives - when I run on a map I have made with KP I have learned to read the contours and use form lines. Pretty much how you when you were younger had to learn different mappers style.

Today the most common thing I hear from not so used orienteerers are that they feel the base contour should have been drawn where the formlines sometimes end up, which I can agree to when I stand in the forrest and draw competition maps. And when on location, as a mapper - sometimes you have to manually connect contours that are at significant changes with others - and have contours that shift between the "glided" possitions of the contours.

But what I understand from you reply is that the contours from KP are pretty spot on where the actual 5 meters intervalls, or whatever setting you have? Or do they in some way actually look for if there are more changes a littele bit higher or lower and adjust itself to that?

**Reply by Jagge** [reply]
yes, contours are actual 5 m intervals except when various smoothings kicks in or when there is knolls close by. This is why you usually need/get more form lines - human mapper can capture slope's shape with just 5m by gliding controus but KP kind of detects that but instead of gliding triggers form lines to be drawn between full controus. Steeper it is, less there is need for gliding beause a land form usually is more then 2.5m vertically and gets captures somehow. When it is flat, stuff much more easily falls vertically between 2.5m and is not mapped at all unless it glides controus to match the elevation of the land form.

**Reply by Jagge** [reply]
So it goes like this. It detects a knoll too big to be unmapped that falls vertically between interval (usually deafult 2.5m). It decides to map it with the upper full contour, but it can't simply draw it with it and be done with it, because often knolls are surronded by other landforms equally high or even higher than the knoll. The map would end up confusing with just the knoll mapped but not the surrounding spurs, so it gently glides all the contours close to the knoll, so that would it capture those neigboring  landforms too one way or an other. See, to some extent orienteers understand everything can not be mapped and something can fall between contours, but it is much harder to understand something is not mapped even if it is right at the elevation of the contour (knoll being mapped but spurs/reentrants next to it are not). But such gliding is not free, it has side effets, contours will have to slowly go to their original elevation and that means we may map there something that dows not exists. For slopes I thoght it's safer to simply rely on contour interval and triggering form lines when there is something on instead of tweaking / sliding contours and accepting slightly unpredictable side effects.
![kuva](https://github.com/user-attachments/assets/9bbe01ad-024f-4835-8da4-8a76ad1744ec)
